climada/util/yearsets.py

```diff
@@ -0,0 +1,266 @@
+This file is part of CLIMADA.
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU Lesser General Public License as published by the Free
+Software Foundation, version 3.
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+You should have received a copy of the GNU Lesser General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+---
+Define functions to handle impact year_sets
+"""
+
+import copy
+import logging
+import numpy as np
+from numpy.random import default_rng
+
+import climada.util.dates_times as u_dt
+
+LOGGER = logging.getLogger(__name__)
+
+def impact_yearset(eis, n_resampled_years=None, year_list=None, multiple_events=True,
+                   sampling_vect=None, n_events_per_year=None):
+
+    """PURPOSE:
+      Create an annual impact set (ais) by sampling events for each year from an existing
+      probabilistic event impact set (eis).
+
+    INPUTS:
+      eis (impact class object): an event impact set (eis)
+    OPTIONAL INPUT:
+        n_resampled_years(int): the target number of years the impact yearset shall
+            contain.
+        year_list (list): list of years for the resulting annual impact set
+            (by default a list starting on the 01-01-0001 is generated)
+        multiple_events (boolean): True if the hazard causing the given impact can occur
+            several times per year (such as several tropical cyclones); False for hazards
+            that compute the impact on an annual scale (such as relative cropyield)
+        sampling_vect (array): the sampling vector, technical, see code (can be used to
+          re-create the exact same yearset). Needs to be obtained in a first
+          call, i.e. [ais, sampling_vect] = climada_eis2ais(...) and then
+          provided in subsequent calls(s) to obtain the exact same sampling
+          structure of yearset, i.e ais = climada_eis2ais(..., sampling_vect)
+        n_events_per_year (array): amount of resampled events per year
+            (length = n_resampled_years), can be reused similar to the sampling_vect
+            (to do: combine these two in one variable?)
+    OUTPUTS:
+      ais: the year impact set (ais), a struct with same fields as eis (such
+          as Value, ed, ...). All fields same content as in eis, except:
+          date(i): the year i
+          impact(i): the sum of impact for year(i).
+          frequency(i): the annual frequency, =1
+      sampling_vect: the sampling vector, technical, see code (can be used to
+                     re-create the exact same yearset)
+      """
+
+    # OLD CODE - STILL NEEDED?
+    # if years are known:
+    # events = list(EIS.calc_impact_year_set().keys())
```

Comment:
Why did you decide to not use the already existing method for computing the impact year set for a given set of events?

Reply:
I think I wanted to use this for hazards that have 1 event per year (otherwise sampling on impact_year_set basis doesn't make much sense, right?). I guess we can delete it.